tests/adapters/switches/juniper_qfx_copper_test.py

```diff
@@ -651,3 +653,69 @@ def test_change_bond_speed_update_slaves_and_interface_at_same_time(self):
+                      <l2ng-l2-mac-entry-flags>in_hash,in_ifd,in_ifl,in_vlan,in_rtt,kernel,in_ifbd</l2ng-l2-mac-entry-flags>
+                      <l2ng-l2-mac-epoch>1</l2ng-l2-mac-epoch>
+                      <l2ng-l2-mac-sequence-number>2</l2ng-l2-mac-sequence-number>
+                      <l2ng-l2-mac-learn-mask>0x00000002</l2ng-l2-mac-learn-mask>
+                    </l2ng-l2ald-mac-entry-vlan>
+                    <l2ng-l2ald-mac-entry-vlan style="extensive">
+                      <l2ng-l2-mac-address>00:11:90:3d:33:58</l2ng-l2-mac-address>
+                      <mac-count-global>6</mac-count-global>
+                      <learnt-mac-count>6</learnt-mac-count>
+                      <l2ng-l2-mac-routing-instance>default-switch</l2ng-l2-mac-routing-instance>
+                      <l2ng-l2-vlan-id>4080</l2ng-l2-vlan-id>
+                      <l2ng-l2-mac-vlan-name>VLAN4080</l2ng-l2-mac-vlan-name>
+                      <l2ng-l2-mac-logical-interface>xe-0/0/6.0</l2ng-l2-mac-logical-interface>
+                      <l2ng-l2-mac-ifl-generation>3045</l2ng-l2-mac-ifl-generation>
+                      <l2ng-l2-mac-entry-flags>in_hash,in_ifd,in_ifl,in_vlan,in_rtt,kernel,in_ifbd</l2ng-l2-mac-entry-flags>
+                      <l2ng-l2-mac-epoch>31</l2ng-l2-mac-epoch>
+                      <l2ng-l2-mac-sequence-number>0</l2ng-l2-mac-sequence-number>
+                      <l2ng-l2-mac-learn-mask>0x00000001</l2ng-l2-mac-learn-mask>
+                    </l2ng-l2ald-mac-entry-vlan>
+                    <l2ng-l2ald-mac-entry-vlan style="extensive">
+                      <l2ng-l2-mac-address>00:11:b7:b4:74:2c</l2ng-l2-mac-address>
+                      <mac-count-global>8</mac-count-global>
+                      <learnt-mac-count>8</learnt-mac-count>
+                      <l2ng-l2-mac-routing-instance>default-switch</l2ng-l2-mac-routing-instance>
+                      <l2ng-l2-vlan-id>4063</l2ng-l2-vlan-id>
+                      <l2ng-l2-mac-vlan-name>VLAN4063</l2ng-l2-mac-vlan-name>
+                      <l2ng-l2-mac-logical-interface>ae47.0</l2ng-l2-mac-logical-interface>
+                      <l2ng-l2-mac-ifl-generation>240</l2ng-l2-mac-ifl-generation>
+                      <l2ng-l2-mac-entry-flags>in_hash,in_ifd,in_ifl,in_vlan,in_rtt,kernel,in_ifbd</l2ng-l2-mac-entry-flags>
+                      <l2ng-l2-mac-epoch>1</l2ng-l2-mac-epoch>
+                      <l2ng-l2-mac-sequence-number>2</l2ng-l2-mac-sequence-number>
+                      <l2ng-l2-mac-learn-mask>0x00000002</l2ng-l2-mac-learn-mask>
+                    </l2ng-l2ald-mac-entry-vlan>
+                </l2ng-l2ald-rtb-macdb>
+                """)))
+        mac_addresses = self.switch.get_mac_addresses()
+        assert_that(len(mac_addresses), is_(3))
+
+        for mac_address in mac_addresses:
+            if mac_address.mac_address == '00:11:c6:01:53:a7':
+                assert_that(mac_address.interface, is_('Ag 47.0'))
```

Comment:
It's weird for me to rename the interface.
Reasons: 
1-  when an engineer knows it a juniper, It will be suprised of seing "Ag" and will get confused that "Ag47.0" as he is not used to see those ports.
2- We don't do this replacement for interface in the /interfaces endpoint, we use whatever we're getting. po, vlan, Ge / GI / xe / FA ... So if you take one interface that is listed there, you won't be able to reuse it on the /interfaces endpoint. You have to do the translation back.

What I suggest:
Letting "interface" field be whatever the interface returns.
Adding a new field for the type of interface that allows you to discriminate. (eg. interface_type = "aggregated" with constants or interface_type="vlan".) the interface won't be changed and you will be able to discriminate in NICO.
This evolves better as we could add something to split the interface into speed /  stack / port or something like that that would be uniform amongst all switches.

Reply:
Another option would be to add the original interface name in a second field "interface_raw".
or rename the current interface name to "interface_normalized"